Disable spam log in yadro-storage-manager service

surf148 opened the pull request:
storage: the yadro-storage-manager service spams many error and debug messages. This commit excludes a spam of error messages. And appends option '--verbose' for disable debug messages.

Signed-off-by: Evgeniy Kislenok e.kislenok@yadro.com
End-user-impact: none

Commit:
Disable spam log in yadro-storage-manager service

storage: the yadro-storage-manager service spams many error and
debug messages. This commit excludes a spam of error messages.

Signed-off-by: Evgeniy Kislenok e.kislenok@yadro.com
End-user-impact: none

nest1ing commented on src/common_i2c.cpp:
```diff
@@ -290,6 +311,33 @@ int i2cDev::i2c_transfer(uint8_t tx_len, uint8_t* tx_data, uint8_t rx_len,
     return res;
 }
 
+bool i2cDev::isSpamingToLog(int res, std::stringstream& ss)
```
I dislike that logging functionality moved from the calling side (by passing `ss`) and prefer this method to keep silence or log the self owned warnings only.

surf148 replied:
fixed

nest1ing commented on src/common_i2c.cpp:
```diff
@@ -290,6 +311,33 @@ int i2cDev::i2c_transfer(uint8_t tx_len, uint8_t* tx_data, uint8_t rx_len,
+        log<level::ERR>(ss.str().c_str());
+    }
+
+    if (numLogErrors < std::numeric_limits<uint32_t>::max())
```
Looks weird.
Likely it would be more readable:
```cpp
if (numLogErrors >= maxLogErrors)
{
        ss << "... (Detected multiple errors. Stoping spam to log.)";
        log<level::ERR>(ss.str().c_str());
}
else
{
        bymLegErrors++;
}
```

surf148 replied:
reworked

nest1ing reviewed (commented):
- What's the reason of dropping the old and creating the new PR?
   Please do not do it without a strong reason.
   This makes the code review more complicated.
- I would prefer to split the adding `verbose` and `isSpammingToLog` onto two separate commits.

Commit:
Add option '--verbose' for enable debug messages

storage: the yadro-storage-manager service always spams many
debug messages. This commit allow to turn on/off debug messages
via service option '--verbose'.

Signed-off-by: Evgeniy Kislenok e.kislenok@yadro.com
End-user-impact: none

surf148 commented:
> * What's the reason of dropping the old and creating the new PR?
>   Please do not do it without a strong reason.
>   This makes the code review more complicated.
> * I would prefer to split the adding `verbose` and `isSpammingToLog` onto two separate commits.
